### PR TITLE
Adapt to TranslationProject Split & add type chips to UI

### PR DESCRIPTION
**Pull request by willdch**
API PR: https://github.com/SeedCompany/cord-api-v3/pull/3160

## Description

Added a chip displaying the project type on the Project Card, Project Overview Page and Partner table project tab. 

Project Card:
Screenshot 2024-04-08 at 3 44 50 PM

Project Overview:
Screenshot 2024-04-08 at 3 45 11 PM

Partner Table Project tab:
Screenshot 2024-04-09 at 11 19 29 AM

- [x] Change the task url above to the actual Monday task
- [ NA] Add/update tests if needed
- [x] Add reviewers to this PR

### CODE REVIEW

File: src/components/ProjectListItemCard/ProjectListItem.graphql [hunk 1]
```diff
@@ -28,7 +28,12 @@ fragment ProjectListItem on Project {
       total
     }
   }
-  ... on TranslationProject {
+  ... on MomentumProject {
+    engagements {
+      total
+    }
+  }
+  ... on MultiplicationProject {
```
Comment (CarsonF) [inline src/components/ProjectListItemCard/ProjectListItem.graphql:36]: There's no need to intersect this on concretes. The Project interface has this field.
```gql
...TogglePin
engagements { total }
```

File: src/components/form/Lookup/Project/ProjectLookup.graphql [hunk 2]
```diff
@@ -1,6 +1,9 @@
 query ProjectLookup($query: String!) {
   search(
-    input: { query: $query, type: [TranslationProject, InternshipProject] }
+    input: {
+      query: $query
+      type: [MomentumProject, MultiplicationProject, InternshipProject]
```
Comment (CarsonF) [inline src/components/form/Lookup/Project/ProjectLookup.graphql:5]: I think we can just use Project here.
```suggestion
      type: [Project]
```

File: src/components/form/Lookup/Project/ProjectLookup.graphql [hunk 3]
```diff
@@ -11,10 +14,15 @@ query ProjectLookup($query: String!) {
 }
 
 query TranslationProjectLookup($query: String!) {
-  search(input: { query: $query, type: [TranslationProject] }) {
+  search(
+    input: { query: $query, type: [MomentumProject, MultiplicationProject] }
```
Comment (CarsonF) [inline src/components/form/Lookup/Project/ProjectLookup.graphql:18]: The rest of this file shouldn't need to change.

File: src/scenes/Engagement/Delete/DeleteEngagement.tsx [hunk 4]
```diff
@@ -19,7 +20,8 @@ import {
 } from './DeleteEngagement.graphql';
 
 type ProjectIdFragment =
-  | TranslationProjectIdFragment
+  | MomentumProjectIdFragment
+  | MultiplicationProjectIdFragment
   | InternshipProjectIdFragment;
```
Comment (CarsonF) [inline src/scenes/Engagement/Delete/DeleteEngagement.tsx:25]: I'd move this type alias to a common spot. Probably in `common/fragments/` somewhere.

File: src/scenes/Engagement/LanguageEngagement/Create/CreateLanguageEngagement.graphql [hunk 5]
```diff
@@ -6,7 +6,7 @@ mutation createLanguageEngagement($input: CreateLanguageEngagementInput!) {
   }
 }
 
-fragment TranslationProjectSensitivity on TranslationProject {
+fragment TranslationProjectSensitivity on MomentumProject {
```
Comment (CarsonF) [inline src/scenes/Engagement/LanguageEngagement/Create/CreateLanguageEngagement.graphql:9]: Revert

File: src/scenes/Projects/ChangeRequest/Update/UpdateProjectChangeRequest.tsx [hunk 6]
```diff
@@ -29,7 +29,10 @@ import {
 
 export interface UpdateProjectChangeRequestFormParams {
   project: {
-    __typename?: 'TranslationProject' | 'InternshipProject';
+    __typename?:
+      | 'MomentumProject'
+      | 'MultiplicationProject'
+      | 'InternshipProject';
     id: string;
   };
```
Comment (CarsonF) [inline src/scenes/Projects/ChangeRequest/Update/UpdateProjectChangeRequest.tsx:37]:
```ts
project: ProjectIdFragement
```

File: src/scenes/Projects/List/ProjectList.tsx [hunk 7]
```diff
@@ -56,7 +56,8 @@ export const ProjectList = () => {
       input: {
         ...sort.value,
         filter: {
-          ...omit(filters, 'tab'),
+          type: filters.type ? [filters.type] : undefined,
+          ...omit(filters, 'tab', 'type'),
```
Comment (CarsonF) [inline src/scenes/Projects/List/ProjectList.tsx:60]: What's this about?

**Review by CarsonF — CHANGES_REQUESTED** [review]

File: src/scenes/Projects/List/ProjectList.tsx (hunk 7 above)
Comment (andrewmurraydavid) [reply src/scenes/Projects/List/ProjectList.tsx:60]: You made the `type` an array on the API and that broke the filter as it's expecting an array. To keep the radio/single select functionality, we're sending the type value wrapped in a list

File: src/scenes/Engagement/LanguageEngagement/Create/CreateLanguageEngagement.graphql (hunk 5 above)
Comment (andrewmurraydavid) [reply src/scenes/Engagement/LanguageEngagement/Create/CreateLanguageEngagement.graphql:9]: needs concrete

File: src/scenes/Projects/List/ProjectList.tsx (hunk 7 above)
Comment (CarsonF) [reply src/scenes/Projects/List/ProjectList.tsx:60]: Let's just change these two spots to match the new API instead
https://github.com/SeedCompany/cord-field/blob/22709b16472c1fc73043dcaacdf4915aa879584d/src/scenes/Projects/List/ProjectFilterOptions.tsx#L22-L22
https://github.com/SeedCompany/cord-field/blob/22709b16472c1fc73043dcaacdf4915aa879584d/src/scenes/Projects/List/ProjectFilterOptions.tsx#L47-L48

**Comment by CarsonF** [comment]
My only remaining comment, is that the ProjectOverview uses the default Chip font size which is 13px, but everywhere else (in lists) its explicitly `body2` which is size 14px.

We should pick one size and use that as the default for all chips.

**Comment by andrewmurraydavid** [comment]
> My only remaining comment, is that the ProjectOverview uses the default Chip font size which is 13px, but everywhere else (in lists) its explicitly `body2` which is size 14px.
> 
> We should pick one size and use that as the default for all chips.

@willdch is working on that

**Comment by CarsonF** [comment]
@willdch @andrewmurraydavid I rejiggered the commits here to be single responsibility so we don't need to squash.
I also removed the `body2` size.

Want to give it a final look over before I merge?

**Review by CarsonF — APPROVED** [review]

**Review by andrewmurraydavid — CHANGES_REQUESTED** [review]
There is a caching issue between Momentum and Multiplication. That's what was blocking us from addressing other things as that was a lot bigger priority than small UI tweaks.
Because Momentum and Multiplication are both Translation, Apollo doesn't make a call to fetch when switching between Momentum and Multiplication.

I will need to check if your changes fixed that.

**Comment by CarsonF** [comment]
> Apollo doesn't make a call to fetch when switching between Momentum and Multiplication.

Wdym? how to reproduce?

**Comment by andrewmurraydavid** [comment]
ok, not apollo, but the call to the API doesn't go through because it thinks it's cached.

I said apollo since it's responsible with caching as i recall.

To reproduce, go to the Project list, select Internship, apply and you will see the Internship projects. Then select multiplication and you will see Momentum, not multiplication.

**Comment by andrewmurraydavid** [comment]
I just confirmed, @CarsonF , it's still happening

**Review by andrewmurraydavid — APPROVED** [review]
Thanks for fixing the filter issue, @CarsonF !

**Pull request merged**